server/base/src/main/java/org/apache/accumulo/server/AbstractServer.java

```diff
@@ -139,6 +149,41 @@ public String getApplicationName() {
     return applicationName;
   }
 
+  /**
+   * Get the ServiceLock for this server process. May return null if called before the lock is
+   * acquired.
+   *
+   * @return lock ServiceLock or null
+   */
+  public abstract ServiceLock getLock();
+
+  public void startServiceLockVerificationThread() {
+    final long interval =
```

Comment:
Would the following precondition make sense?  Are there any other expectations that could be checked?

```suggestion
    Preconditions.checkState(verificationThread == null);
    final long interval =
```

Reply:
Added this check and one like it for the serverThread in 601aff5.